src/components/EffectiveDateTime/EffectiveDateTime.vue

```diff
@@ -0,0 +1,477 @@
+  }
+
+  @Watch('currentJsDate', { immediate: true })
+  onCurrentJsDateChanged (val: Date) {
+    // safety check (val may be null)
+    if (val) {
+      // set new min date
+      const minDate = new Date()
+      // add 3 minutes
+      minDate.setTime(val.getTime() + this.MIN_DIFF_MINUTES * 60 * 1000)
+      this.minDate = minDate
+
+      // set new max date
+      const maxDate = new Date()
+      // add 10 days
+      maxDate.setTime(val.getTime() + this.MAX_DIFF_DAYS * 24 * 60 * 60 * 1000)
+      this.maxDate = maxDate
+
+      // check if form is still valid
+      this.emitValid()
+    }
+  }
+
+  @Watch('datePicker')
+  onDatePickerChanged (val: string): void {
+    this.dateText = val
+    // the watcher for dateText will fire next
+  }
+
+  @Watch('dateText')
+  onDateTextChanged (val: string): void {
+    if (this.isFutureEffective) {
+      this.constructAndUpdate()
+    }
+  }
+
+  @Watch('selectHour')
+  onSelectHourChanged (val: string): void {
+    if (this.isFutureEffective) {
+      this.constructAndUpdate()
+    }
+  }
+
+  @Watch('selectMinute')
+  onSelectMinuteChanged (val: string): void {
+    if (this.isFutureEffective) {
+      this.constructAndUpdate()
+    }
+  }
+
+  @Watch('selectPeriod')
+  onSelectPeriodChanged (val: string): void {
+    if (this.isFutureEffective) {
+      this.constructAndUpdate()
+    }
+  }
+
+  @Watch('effectiveDateType')
+  onEffectiveDateTypeChanged (val: EffectiveDateTypes): void {
+    this.isImmediate = (val === EffectiveDateTypes.IMMEDIATE)
+    this.isFutureEffective = (val === EffectiveDateTypes.FUTURE_EFFECTIVE)
+
+    // if we changed to IMMEDIATE then clear the model values (otherwise retain them)
+    if (this.isImmediate) {
+      this.datePicker = ''
+      this.dateText = ''
+      this.selectHour = []
+      this.selectMinute = []
+      this.selectPeriod = PeriodTypes.AM
+      this.$refs.datePickerRef.clearDate()
+    }
+
+    // update the parent
+    this.emitIsFutureEffective(this.isFutureEffective)
+    this.emitEffectiveDate(null)
+    this.emitValid()
+  }
+
+  @Emit('isFutureEffective')
+  private emitIsFutureEffective (val: boolean): void {}
+
+  @Emit('effectiveDate')
+  private emitEffectiveDate (val: Date): void {}
+
+  @Emit('valid')
+  private async emitValid (): Promise<boolean> {
+    // localized dateText check for future effective selections
+    const validDateText = this.isFutureEffective ? !!this.dateText : true
+
+    // wait for form to update itself before checking validity
+    await Vue.nextTick()
+    const isDateValid = this.$refs.datePickerRef.validateForm()
+    const isTimeValid = this.$refs.form.validate()
+    return this.isImmediate || (!!this.effectiveDateType &&
+      isDateValid && isTimeValid &&
+      !!this.selectHour.length && !!this.selectMinute.length &&
+      !this.isUnderTime &&
+      !this.isOverTime &&
+      validDateText
+    )
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
```

Comment:
Related to the other question, is this something we have defined here? or could this present an issue if pulled in and use in a project without this hierarchy.

Reply:
Yes, if a parent project is missing this folder/file, this will fail. But it has always been this way for all our projects.

All our projects (except NR UI) use the folders above.

Reply:
Ok cool